### PR TITLE
[Enhancement] Optimize creation of scan mv in register mv

### LINKED ISSUE #13969: [Enhancement] the creation of scan mv in register mv can be optimized
## Enhancement
the creation of scan mv in register mv during mv query rewrite can be optimized.
originally the creation of scan mv is based on sql, which can be enhanced by constructing LogicalOlapScanOperator directly.

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13969 

## Problem Summary(Required) ：

The creation of scan mv in register mv during mv query rewrite can be optimized.
Originally the creation of scan mv is based on sql,which can be enhanced
by constructing LogicalOlapScanOperator directly.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java [hunk 1]
```diff
@@ -226,36 +238,100 @@ private void registerMaterializedViews(OptExpression logicOperatorTree, ConnectC
...
+        List<Long> selectPartitionIds = Lists.newArrayList();
+        Set<String> excludedPartitions = mv.getPartitionNamesToRefreshForMv();
+        List<String> selectedPartitionNames = mv.getPartitionNames()
+                .stream().filter(name -> !excludedPartitions.contains(name)).collect(Collectors.toList());
+        for (Partition p : mv.getPartitions()) {
+            if (selectedPartitionNames.contains(p.getName())) {
+                selectPartitionIds.add(p.getId());
+            }
+        }
+        PartitionNames partitionNames = new PartitionNames(false, selectedPartitionNames);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java:320]: Mv scan with **not**  out of date partition, it still need use base table partition range as predicate?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java [hunk 2]
```diff
@@ -226,36 +238,100 @@ private void registerMaterializedViews(OptExpression logicOperatorTree, ConnectC
...
-            if (!MvUtils.isLogicalSPJ(scanMvPlan)) {
-                continue;
+            List<ColumnRefOperator> mvOutputColumns = mvOptimizer.getOutputExpressions();
+            MaterializationContext materializationContext =
+                    new MaterializationContext(mv, mvPlan, queryColumnRefFactory, mvColumnRefFactory);
+            // generate scan mv plan here to reuse it in rule applications
+            LogicalOlapScanOperator scanMvOp = createScanMvExpression(materializationContext);
+            materializationContext.setScanMvOperator(scanMvOp);
+            String dbName = connectContext.getGlobalStateMgr().getDb(mv.getDbId()).getFullName();
+            connectContext.getDumpInfo().addTable(dbName, mv);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java:255]: We do not support dump for mv now

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java (hunk 1 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java:320]: no, it is not necessary to add partition range predicate by using the method.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 95 / 100 (95.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/PlannerProfile.java|0|2|00.00%|[221, 222]|
|:large_blue_circle:|com/starrocks/sql/optimizer/MvRewritePreprocessor.java|81|84|96.43%|[59, 70, 78]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/MaterializationContext.java|6|6|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**